### PR TITLE
fix useExplicitVersion

### PR DESCRIPTION
**Pull request by pjain1**
regression in https://github.com/druid-io/druid/pull/3554

### CODE REVIEW

File: indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java [hunk 1]
```diff
@@ -204,15 +204,17 @@ public TaskStatus run(TaskToolbox toolbox) throws Exception
...
-      version = specVersion;
-    } else {
-      log.error(
-          "Spec version can not be greater than lock version, Spec version: [%s] Lock version: [%s].",
-          specVersion,
-          version
-      );
-      return TaskStatus.failure(getId());
+    if (indexerSchema.getTuningConfig().isUseExplicitVersion()) {
+      if (specVersion.compareTo(version) < 0) {
```
Comment (akashdw) [inline indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java]: null/empty check for `specVersion`?  where `isUseExplicitVersion` is set to true but `specVersion` is missing in HadoopIngestionSpec.

Comment (pjain1) [inline indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java]: I think version cannot be null, see this - https://github.com/druid-io/druid/blob/master/indexing-hadoop/src/main/java/io/druid/indexer/HadoopTuningConfig.java#L115

**Comment by cheddar** [comment]
👍

**Comment by akashdw** [comment]
👍 after travis.

**Comment by himanshug** [comment]
:+1:

**Pull request merged**